### PR TITLE
Cleanup the usage of `layer_norm_eps` in some models

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

**(No breaking change in this PR)**
**(So far I only change `OneFormerConfig`, but I will update other config classes whose default `layer_norm_eps = 1e-5`)**

Fix the missing usage of `config.layer_norm_eps` in some pytorch models: **if the default value in the config class is `1e-05`** (i.e. the same as the default value in `nn.LayerNorm`. In this case, we don't break the current behavior, and **make the (WIP) test dealing with much fewer edge cases** (which is always a bit of burden).

**In #20699, my claim regarding the breaking change was a bit misleading**: only those config classes having default `layer_norm_eps` not being `1e-5` (for example, `LxmertConfig`) will have breaking change - which I don't do anything in this PR.

### CODE REVIEW

**Comment by ydshieh** [comment]
@sgugger @NielsRogge  Just want to hear from you to make sure we are good for this change 🙏

File: src/transformers/models/oneformer/modeling_oneformer.py [hunk 1]
```diff
@@ -1633,11 +1633,13 @@ def forward(
 
 
 class OneFormerTransformerDecoderSelfAttentionLayer(nn.Module):
-    def __init__(self, embed_dim, num_heads, dropout=0.0, activation="relu", normalize_before=False):
+    def __init__(
+        self, embed_dim, num_heads, dropout=0.0, activation="relu", normalize_before=False, layer_norm_eps=1e-05
```
Comment (ydshieh) [inline src/transformers/models/oneformer/modeling_oneformer.py:1637]: We don't have `config` argument here, so I need to add `layer_norm_eps`. The `config.layer_norm_eps` will be passed from the module(s) that contain(s) this module.

**Review by sgugger — COMMENTED** [review]
This should be done on a case-by-case basis. If the model uses the same as the PyTorch default, I have no problem adding the esp as you did. For others, it might break the forward pass, which is normally compared to the original implementation.

File: src/transformers/models/altclip/configuration_altclip.py [hunk 2]
```diff
@@ -173,8 +173,9 @@ class AltCLIPVisionConfig(PretrainedConfig):
             The size (resolution) of each patch.
         hidden_act (`str` or `function`, *optional*, defaults to `"quick_gelu"`):
             The non-linear activation function (function or string) in the encoder and pooler. If string, `"gelu"`,
-            `"relu"`, `"selu"` and `"gelu_new"` ``"quick_gelu"` are supported. layer_norm_eps (`float`, *optional*,
```
Comment (ydshieh) [inline src/transformers/models/altclip/configuration_altclip.py:176]: docstring is broken here and in several other places.

File: src/transformers/models/altclip/modeling_altclip.py [hunk 3]
```diff
@@ -844,9 +844,9 @@ def __init__(self, config: AltCLIPConfig):
         super().__init__()
         self.embed_dim = config.hidden_size
         self.self_attn = AltCLIPAttention(config)
-        self.layer_norm1 = nn.LayerNorm(self.embed_dim)
+        self.layer_norm1 = nn.LayerNorm(self.embed_dim, eps=config.layer_norm_eps)
```
Comment (ydshieh) [inline src/transformers/models/altclip/modeling_altclip.py:847]: A few lines above, we have `def __init__(self, config: AltCLIPConfig):`. This is incorrect. It should be `AltCLIPVisionConfig`. Same comment applies for `CLIP`, where it should be `the union of CLIPTextConfig and CLIPVisionConfig`.

I decide not to address this issue in this PR though - a bit too much involved.

File: src/transformers/models/blip/configuration_blip.py [hunk 4]
```diff
@@ -247,7 +249,7 @@ def __init__(
         image_size=384,
         patch_size=16,
         hidden_act="gelu",
-        layer_norm_eps=0.00001,
+        layer_norm_eps=1e-5,
```
Comment (ydshieh) [inline src/transformers/models/blip/configuration_blip.py:252]: just make it easier to read.

File: src/transformers/models/x_clip/modeling_x_clip.py [hunk 5]
```diff
@@ -1218,8 +1218,8 @@ def __init__(self, config):
 
         embed_dim = config.projection_dim
         self.cross_attn = XCLIPCrossAttention(config)
-        self.norm1 = nn.LayerNorm(embed_dim)
-        self.norm3 = nn.LayerNorm(embed_dim)
+        self.norm1 = nn.LayerNorm(embed_dim, eps=config.text_config.layer_norm_eps)
```
Comment (ydshieh) [inline src/transformers/models/x_clip/modeling_x_clip.py:1221]: For the change here and below, I tried to read the source code and use the `text_config` and `vision_config` depending on the context. (Although, if `eps` gets the same value in these 2 configs, either could work.)

cc @NielsRogge to see if you have any comment.

**Review by ydshieh — COMMENTED** [review]
Only change (some of) the models with `1e-5` as the default value for `layer_norm_eps` in the config classes.

Also:
- fix a few broken docstring
- change `0.00001` to `1e-5`

File: src/transformers/models/clipseg/configuration_clipseg.py [hunk 6]
```diff
@@ -270,8 +272,7 @@ class CLIPSegConfig(PretrainedConfig):
             The dropout ratio for the attention probabilities.
         decoder_hidden_act (`str` or `function`, *optional*, defaults to `"quick_gelu"`):
             The non-linear activation function (function or string) in the encoder and pooler. If string, `"gelu"`,
-            `"relu"`, `"selu"` and `"gelu_new"` ``"quick_gelu"` are supported. layer_norm_eps (`float`, *optional*,
-            defaults to 1e-5): The epsilon used by the layer normalization layers.
```
Comment (ydshieh) [inline src/transformers/models/clipseg/configuration_clipseg.py:274]: remove `layer_norm_eps` for the top level `CLIPSegConfig`, as it is never used, and to be aligned with other CLIP-like models

**Review by sgugger — APPROVED** [review]
Thanks for all the fixes!

**Pull request merged**